compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -476,6 +487,26 @@ public boolean checkStructEquivalency(BType rhsType, BType lhsType) {
         return errorTypes;
     }
 
+    public BType inferRecordFieldType(BRecordType recordType) {
+        boolean isSameType = true;
+        List<BField> fields = recordType.fields;
+        BType inferredType = fields.get(0).type; // If all the fields are the same, doesn't matter which one we pick
+
+        for (int i = 1; i < fields.size(); i++) {
+            isSameType = isSameType && fields.get(i - 1).type.tag == fields.get(i).type.tag;
```

Comment:
do we need the "isSameType" check here? We only need to check if all consecutive fields are of the same type (starting with the 0th index field) or if each field's type is "inferredType" right?

Reply:
True, will change it